Brigthlocal/Api.cs

```diff
@@ -0,0 +1,155 @@
+﻿using Newtonsoft.Json;
+using RestSharp;
+using System;
+using System.Collections.Generic;
+using System.Dynamic;
+using System.Security.Cryptography;
+using System.Web;
+
+namespace Brightlocal
+{
+
+    public class Api
+    {
+        /** API endpoint URL */
+        const string ENDPOINT = "https://tools.brightlocal.com/seo-tools/api";
+        /** expiry can't be more than 30 minutes (1800 seconds) */
+        const int MAX_EXPIRY = 1800;
+
+        private Uri endpoint = new Uri("https://tools.brightlocal.com/seo-tools/api/");
+        private string apiKey;
+        private string apiSecret;
+
+        // create an instance of restsharp client
+        private readonly RestClient client = new RestClient();
+
+
+        public Api(string apiKey, string apiSecret, string endpoint = null)
+        {
+            this.apiKey = apiKey;
+            this.apiSecret = apiSecret;
+            if (endpoint != null && Uri.IsWellFormedUriString(endpoint, UriKind.Absolute))
+            {
+                this.endpoint = new Uri(endpoint);
+            }
+        }
+
+        public IRestResponse Get(string resource, Parameters parametrs)
+        {
+            return Call(resource, parametrs, Method.GET);
+        }
+
+        public IRestResponse Post(string resource, Parameters parametrs)
+        {
+            return Call(resource, parametrs, Method.POST);
+        }
+
+        public IRestResponse Delete(string resource, Parameters parametrs)
+        {
+            return Call(resource, parametrs, Method.DELETE);
+        }
+
+        public IRestResponse Put(string resource, Parameters parametrs)
+        {
+            return Call(resource, parametrs, Method.PUT);
+        }
+
+        public Batch CreateBatch(bool stopOnJobError = false, string callbackUrl = null)
+        {
+            Parameters parametrs = new Parameters();
+            parametrs.Add("stop-on-job-error", stopOnJobError);
+            if (callbackUrl != null)
+            {
+                parametrs.Add("callback", callbackUrl);
+            }
+            IRestResponse response = Post("/v4/batch", parametrs);
+            dynamic content = JsonConvert.DeserializeObject(response.Content);
+            if (content.success != true)
+            {
+                throw new ApplicationException("An error occurred and we weren\'t able to create the batch. " + content.errors, content.ErrorException);                
```

Comment:
Why ApplicationException? If you can't do a successful Post to 'batch' endpoint, return smth meaningful to the client.
https://docs.microsoft.com/en-us/dotnet/api/system.applicationexception?view=net-5.0